src/elements/content-sharing/ContentSharing.js

```diff
@@ -110,30 +145,122 @@ function ContentSharing({ apiHost, displayInModal, itemID, itemType, language, t
+            setItem(prevItem => ({ ...prevItem, ...updatedItem }));
+            setSharedLink(prevSharedLink => ({ ...prevSharedLink, ...updatedSharedLink }));
+        };
+
+        // Handle failed PUT requests to /files or /folders
+        const handleUpdateItemError = () => {
+            const updatedNotifications = { ...notifications };
+            updatedNotifications[notificationID] = createNotification(
+                STATUS_ERROR,
+                contentSharingMessages.sharedLinkUpdateError,
+            );
+            setNotifications(updatedNotifications);
+            setNotificationID(notificationID + 1);
+        };
+
+        if (item && itemPermissions && !onAddLink) {
+            const dataForAPI = {
+                id: itemID,
+                permissions: itemPermissions,
+            };
+
+            let itemAPIInstance;
+            if (itemType === TYPE_FILE) {
+                itemAPIInstance = api.getFileAPI();
+            } else if (itemType === TYPE_FOLDER) {
+                itemAPIInstance = api.getFolderAPI();
+            }
+
+            const updatedOnAddLink: SharedLinkUpdateFnType = () => () =>
+                itemAPIInstance.share(
+                    dataForAPI,
+                    ACCESS_COLLAB,
+                    handleUpdateItemSuccess,
+                    handleUpdateItemError,
+                    CONTENT_SHARING_SHARED_LINK_UPDATE_PARAMS,
+                );
+            setOnAddLink(updatedOnAddLink);
+        }
+    }, [
+        api,
+        createNotification,
+        item,
+        itemID,
+        itemPermissions,
+        itemType,
+        notificationID,
+        notifications,
+        onAddLink,
+        setOnAddLink,
+        sharedLink,
+    ]);
+
+    if (componentErrorMessage) {
+        return <ErrorMask errorHeader={<FormattedMessage {...componentErrorMessage} />} />;
     }
 
     if (item && sharedLink) {
         return (
-            <Internationalize language={language} messages={usmMessages}>
-                <UnifiedShareModal
-                    canInvite={sharedLink.canInvite}
-                    changeSharedLinkPermissionLevel={() => Promise.resolve([])} // to do: replace with a POST to the Shared Link API
-                    collaboratorsList={{ collaborators: [] }} // to do: replace with Collaborators API
-                    currentUserID={currentUserID}
-                    displayInModal={displayInModal}
-                    getCollaboratorContacts={() => Promise.resolve([])} // to do: replace with Collaborators API
-                    getSharedLinkContacts={() => Promise.resolve([])} // to do: replace with Collaborators API
-                    initialDataReceived
-                    item={item}
-                    onAddLink={() => Promise.resolve([])} // to do: replace with a POST to the Shared Link API
-                    sharedLink={sharedLink}
-                />
-            </Internationalize>
+            <>
```

Comment:
It's a bit unconventional for an element to return more than one node. It seems like we're trying to do a bit too much in one component. Should the notifications live elsewhere?